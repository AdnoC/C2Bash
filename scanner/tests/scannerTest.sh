wd="$(dirname "${BASH_SOURCE[0]}")"
source "$wd/../../main.sh"
export __DEBUG=true 

output="$(source "$C2BASH_HOME"/scanner/scanner.sh "$wd/sample.c")"
if echo "$output" | diff "$wd"/scannerTest.expected - ; then
  echo "Test Succeeded"
  return 0
else
  echo "Test Failed"
  return 1
fi
