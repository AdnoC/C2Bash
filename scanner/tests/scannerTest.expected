   Scanning File
    T_IDENTIFIER -> main
    T_WHITESPACE -> 	
    T_LPAREN -> (
    T_RPAREN -> )
    T_LBRACE -> <%
    T_IDENTIFIER -> printf
    T_LPAREN -> (
    T_STRING_CONSTANT -> "Hello Worldn"
    T_RPAREN -> )
    T_SEMICOLON -> ;
    T_IDENTIFIER -> printf
    T_LPAREN -> (
    T_STRING_CONSTANT -> ""
    T_RPAREN -> )
    T_SEMICOLON -> ;
    T_INT -> int
    T_WHITESPACE ->  
    T_IDENTIFIER -> i
    T_WHITESPACE ->  
    T_ASSIGN_OP -> =
    T_WHITESPACE ->  
    T_INTEGER_CONSTANT -> 0
    T_SEMICOLON -> ;
    T_IDENTIFIER -> i
    T_INCR -> ++
    T_SEMICOLON -> ;
    T_INT -> int
    T_WHITESPACE ->  
    T_IDENTIFIER -> i2
    T_WHITESPACE ->  
    T_ASSIGN_OP -> =
    T_WHITESPACE ->  
    T_INTEGER_CONSTANT -> 212341
    T_SEMICOLON -> ;
    T_INT -> int
    T_WHITESPACE ->  
    T_IDENTIFIER -> a
    T_LBRACKET -> [
    T_INTEGER_CONSTANT -> 3
    T_RBRACKET -> ]
    T_SEMICOLON -> ;
    T_LPAREN -> (
    T_STAR -> *
    T_IDENTIFIER -> a
    T_RPAREN -> )
    T_ASSIGN_OP -> =
    T_INTEGER_CONSTANT -> 4
    T_SEMICOLON -> ;
    T_INT -> int
    T_WHITESPACE ->  
    T_STAR -> *
    T_IDENTIFIER -> p
    T_WHITESPACE ->  
    T_ASSIGN_OP -> =
    T_WHITESPACE ->  
    T_LPAREN -> (
    T_INT -> int
    T_WHITESPACE ->  
    T_LBRACKET -> [
    T_RBRACKET -> ]
    T_RPAREN -> )
    T_LBRACE -> {
    T_INTEGER_CONSTANT -> 2
    T_COMMA -> ,
    T_WHITESPACE ->  
    T_INTEGER_CONSTANT -> 4
    T_RBRACE -> }
    T_SEMICOLON -> ;
    T_BOOL -> _Bool
    T_WHITESPACE ->  
    T_IDENTIFIER -> _ident
    T_WHITESPACE ->  
    T_ASSIGN_OP -> =
    T_WHITESPACE ->  
    T_IDENTIFIER -> i
    T_WHITESPACE ->  
    T_EQ -> ==
    T_WHITESPACE ->  
    T_IDENTIFIER -> i2
    T_SEMICOLON -> ;
    T_IDENTIFIER -> i
    T_INCR -> ++
    T_PLUS -> +
    T_IDENTIFIER -> i2
    T_SEMICOLON -> ;
    T_LPAREN -> (
    T_STAR -> *
    T_IDENTIFIER -> a
    T_RPAREN -> )
    T_WHITESPACE ->  
    T_ASSIGN_OP -> =
    T_WHITESPACE ->                                       
    T_INTEGER_CONSTANT -> 2
    T_SEMICOLON -> ;
    T_IDENTIFIER -> printf
    T_LPAREN -> (
    T_STRING_CONSTANT -> "a = %dn"
    T_COMMA -> ,
    T_WHITESPACE ->  
    T_STAR -> *
    T_IDENTIFIER -> a
    T_RPAREN -> )
    T_SEMICOLON -> ;
    T_RBRACE -> }
   Finished scanning
